substrate/bin/node/cli/src/chain_spec.rs

```diff
@@ -345,101 +313,56 @@ pub fn testnet_genesis(
 	root_key: AccountId,
 	endowed_accounts: Option<Vec<AccountId>>,
 ) -> serde_json::Value {
-	let (initial_authorities, endowed_accounts, num_endowed_accounts, stakers) =
+	let (initial_authorities, endowed_accounts, stakers) =
 		configure_accounts(initial_authorities, initial_nominators, endowed_accounts, STASH);
-	const MAX_COLLECTIVE_SIZE: usize = 50;
-
-	let dev_stakers = if cfg!(feature = "staking-playground") {
-		let random_validators =
-			std::option_env!("VALIDATORS").map(|s| s.parse::<u32>().unwrap()).unwrap_or(100);
-		let random_nominators = std::option_env!("NOMINATORS")
-			.map(|s| s.parse::<u32>().unwrap())
-			.unwrap_or(3000);
-		Some((random_validators, random_nominators))
-	} else {
-		None
-	};
 
-	let validator_count = if cfg!(feature = "staking-playground") {
-		std::option_env!("VALIDATOR_COUNT")
-			.map(|v| v.parse::<u32>().unwrap())
-			.unwrap_or(100)
+	let staking_playground_config = if cfg!(feature = "staking-playground") {
+		Some(get_staking_play_ground_config())
 	} else {
-		initial_authorities.len() as u32
+		None
 	};
 
-	let minimum_validator_count =
-		if cfg!(feature = "staking-playground") { 10 } else { initial_authorities.len() as u32 };
-
-	serde_json::json!({
-		"balances": {
-			"balances": endowed_accounts.iter().cloned().map(|x| (x, ENDOWMENT)).collect::<Vec<_>>(),
-		},
-		"session": {
-			"keys": initial_authorities
-				.iter()
-				.map(|x| {
-					(
-						x.0.clone(),
-						x.0.clone(),
-						session_keys(
-							x.2.clone(),
-							x.3.clone(),
-							x.4.clone(),
-							x.5.clone(),
-							x.6.clone(),
-							x.7.clone(),
-						),
-					)
-				})
-				.collect::<Vec<_>>(),
-		},
-		"staking": {
-			"validatorCount": validator_count,
-			"minimumValidatorCount": minimum_validator_count,
-			"invulnerables": initial_authorities.iter().map(|x| x.0.clone()).collect::<Vec<_>>(),
-			"slashRewardFraction": Perbill::from_percent(10),
-			"stakers": stakers.clone(),
-			"devStakers": dev_stakers
-		},
-		"elections": {
-			"members": endowed_accounts
-				.iter()
-				.take(((num_endowed_accounts + 1) / 2).min(MAX_COLLECTIVE_SIZE))
-				.cloned()
-				.map(|member| (member, STASH))
-				.collect::<Vec<_>>(),
-		},
-		"technicalCommittee": {
-			"members": endowed_accounts
-				.iter()
-				.take(((num_endowed_accounts + 1) / 2).min(MAX_COLLECTIVE_SIZE))
-				.cloned()
-				.collect::<Vec<_>>(),
-		},
-		"sudo": { "key": Some(root_key.clone()) },
-		"babe": {
-			"epochConfig": Some(kitchensink_runtime::BABE_GENESIS_EPOCH_CONFIG),
-		},
-		"society": { "pot": 0 },
-		"assets": {
-			// This asset is used by the NIS pallet as counterpart currency.
-			"assets": vec![(9, Sr25519Keyring::Alice.to_account_id(), true, 1)],
-		},
-		"nominationPools": {
-			"minCreateBond": 10 * DOLLARS,
-			"minJoinBond": 1 * DOLLARS,
-		},
-	})
+	kitchensink_runtime::genesis_config_presets::kitchen_sink_genesis(
```

Comment:
Why are we also not using here the runtime function?

Reply:
There is quite a sophisticated even including hardcoded keys for the staging environment, but the presets don't allow setting the account these values.

Hmm, but yeah, now that I think about it, it might make sense to simply add another preset called `staging`. I shall expand the presets. 👍

Reply:
Ah, actually that doesn't work I suppose. The presets require reading from env vars.

```rust
	let random_validators =
		std::option_env!("VALIDATORS").map(|s| s.parse::<u32>().unwrap()).unwrap_or(100);
	let random_nominators = std::option_env!("NOMINATORS")
		.map(|s| s.parse::<u32>().unwrap())
		.unwrap_or(3000);

	let validator_count = std::option_env!("VALIDATOR_COUNT")
		.map(|v| v.parse::<u32>().unwrap())
		.unwrap_or(100);

```

Do you know if this is still used?

Reply:
I think @kianenigma just introduced them recently.

Reply:
True, I will integrate the standard staging setup into the presets, but the staking-playground config reading from env-vars has to remain like this, unfortunately. We have now already two cases, where we would benefit from tweaking the base genesis presets, hence I have created https://github.com/paritytech/polkadot-sdk/issues/7748.

Reply:
I am fine with merging this in the form. 
Personally, I would add some comment explaining that this could be refactored to avoid direct call into native runtime once #7748 is done.

Reply:
Alright, cool. I also favor merging this as is, and then do the refactor after #7748. I have added a comment for that.